pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -1080,4 +1073,129 @@ public void testV3CleanupH3AndTextIndices()
     }
     assertEquals(singleFileIndex.length(), initFileSize);
   }
+
+  @Test
+  public void testV1IfNeedProcess()
+      throws Exception {
+    constructV1Segment();
+    SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(_indexDir);
+    assertEquals(segmentMetadata.getVersion(), SegmentVersion.v1);
+
+    testIfNeedProcess();
```

Comment:
Also add a test where V1 segment has to be preprocessed because IndexLoadingConfig has segmentVersion has V3 ?